pkg/auth/authenticator.go

```diff
@@ -0,0 +1,21 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+type Authenticator interface {
+	Authenticate() error
+	AddAuthenticationHeader(request *http.Request) error
+}
+
+func CreateAuthenticator(config configuration.Configuration, httpClient *http.Client) Authenticator {
+	useOAuth := config.GetBool(configuration.OAuthEnabled)
```

Comment:
Question: I'm not sure this will do what is required, the reason why I'm not sure is, is that I don't understand where OAuthEnabled is being set.